DevTools/Airlock/explainer.md

```diff
@@ -0,0 +1,61 @@
+
+## Status of this feature
+This feature has not yet shipped in Microsoft Edge and is in active development. See the [Changelog section](#changelog) for details on progressive feature implementation.
+
+The team welcomes your feedback on this feature; please [file an issue](https://github.com/MicrosoftEdge/MSEdgeExplainers/issues/new?labels=DevTools&template=devtools.md&title=%5BDevTools%5D+%3CTITLE+HERE%3E) to share your thoughts.
+
+## Introduction
+**Airlock** is a new tool, originally built by the Microsoft Teams team as a DevTools extension to help them find and debug memory leaks. However, given the limited API surface for DevTools extensions, Airlock isn't able to use the [Chromium DevTools Protocol (CDP)](https://chromedevtools.github.io/devtools-protocol/) itself, which forced users of the extension to rely on the **Memory** tool in the Microsoft Edge DevTools to trigger garbage collection (GC) and take heap snapshots. By integrating Airlock as its own tool in the Microsoft Edge DevTools, we can provide innovative, new tooling for debugging DOM node memory leaks.
+
+## Goals
+
+  1. Provide Airlock as a new tool in the Microsoft Edge DevTools to better debug DOM node memory leaks
+
+We're eager to hear your feedback about Airlock and the current state of memory tooling in the Microsoft Edge DevTools. Let us know what you think by filing an [issue on this repo](https://github.com/MicrosoftEdge/MSEdgeExplainers/issues/new?labels=DevTools&template=devtools.md&title=%5BDevTools%5D+%3CTITLE+HERE%3E) or on [Twitter](https://twitter.com/EdgeDevTools).
+
+## Usage
+
+**Note:** Airlock is not currently available in the Microsoft Edge DevTools. The [Changelog section](#changelog) will be updated as soon as a version of Microsoft Edge with Airlock is available.
+
+### What is a DOM node memory leak?
+
+A DOM node memory leak is when a DOM node that is removed from the DOM tree cannot be garbage collected because it is still referenced by the page's JavaScript. Nodes that fall into this category are considered "detached". Depending on the parents and children of the node that causes the leak, a DOM node memory leak can span from just a few nodes all the way to massive trees of hundreds or thousands of elements. This can also be a common occurrence for React apps because React maintains a virtual representation of the DOM in JavaScript.
+<!-- it would be a good idea to host Brian's sample in the EdgeDevToolsSamples repo and point to it from here -->
+
+### Airlock
+
+Airlock displays all of the detached DOM nodes and the size of the tree for each detached node. By triggering GC and querying for the detached nodes in Airlock, you are able to find all the detached nodes that are still allocated in memory. In the following figure, Airlock displays 3 detached nodes, meaning 3 trees of 5 nodes each are still being retained in memory despite being removed from the DOM tree.
+
+![Airlock](airlock.png)
```

Comment:
```suggestion
![Airlock](airlock.png)

Because a **single** detached node can result in **all** related tree nodes being leaked (via references such as `parentNode` or `childNodes`), it can be difficult to determine which specific node within a tree caused the tree to leak.

The **Detach roots** feature destructively disconnects nodes within a tree so they no longer reference one another. When garbage collection is run again, nodes which had no more references to them will be freed, leaving only individual leaky nodes nodes in place. The Memory heap snapshot tool can then be used to identify the offending retaining references to the leaky nodes.
```